File: frontend/src/api/companyProfile.ts
import client from './client';

export interface CompanyProfileResponse {
  id: string;
  agency_id: string;
  company_name: string | null;
  tagline: string | null;
  description: string | null;
  logo_path: string | null;
  logo_url: string | null;
  email: string | null;
  phone: string | null;
  website_url: string | null;
  whatsapp_number: string | null;
  show_phone: boolean;
  show_email: boolean;
  show_website: boolean;
  payment_qr_path: string | null;
  payment_qr_url: string | null;
  payment_note: string | null;
  bank_account_name: string | null;
  bank_name: string | null;
  bank_account_number: string | null;
  bank_ifsc_swift: string | null;
  bank_reference_note: string | null;
  created_at: string;
  updated_at: string;
}

export interface CompanyProfileUpdate {
  company_name?: string | null;
  tagline?: string | null;
  description?: string | null;
  email?: string | null;
  phone?: string | null;
  website_url?: string | null;
  whatsapp_number?: string | null;
  show_phone?: boolean;
  show_email?: boolean;
  show_website?: boolean;
  payment_note?: string | null;
  bank_account_name?: string | null;
  bank_name?: string | null;
  bank_account_number?: string | null;
  bank_ifsc_swift?: string | null;
  bank_reference_note?: string | null;
}

const companyProfileApi = {
  getProfile: async (): Promise<CompanyProfileResponse> => {
    const response = await client.get('/api/v1/company-profile');
    return response.data;
  },

  updateProfile: async (data: CompanyProfileUpdate): Promise<CompanyProfileResponse> => {
    const response = await client.put('/api/v1/company-profile', data);
    return response.data;
  },

  uploadLogo: async (file: File): Promise<CompanyProfileResponse> => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await client.post('/api/v1/company-profile/logo', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  },

  uploadPaymentQR: async (file: File): Promise<CompanyProfileResponse> => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await client.post('/api/v1/company-profile/payment-qr', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  },
};

export default companyProfileApi;

File: backend/app/schemas/company_profile.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class CompanyProfileUpdate(BaseModel):
    """Schema for updating company profile"""
    company_name: Optional[str] = Field(None, max_length=200)
    tagline: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    default_currency: Optional[str] = Field(None, max_length=10)
    accepted_currencies: Optional[str] = Field(None, max_length=255)  # CSV

    email: Optional[str] = Field(None, max_length=200)
    phone: Optional[str] = Field(None, max_length=50)
    website_url: Optional[str] = Field(None, max_length=300)
    whatsapp_number: Optional[str] = Field(None, max_length=50)

    show_phone: Optional[bool] = None
    show_email: Optional[bool] = None
    show_website: Optional[bool] = None

    payment_note: Optional[str] = None
    bank_account_name: Optional[str] = Field(None, max_length=200)
    bank_name: Optional[str] = Field(None, max_length=200)
    bank_account_number: Optional[str] = Field(None, max_length=100)
    bank_ifsc_swift: Optional[str] = Field(None, max_length=50)
    bank_reference_note: Optional[str] = None


class CompanyProfileResponse(BaseModel):
    """Schema for company profile response"""
    id: str
    agency_id: str

    company_name: Optional[str]
    tagline: Optional[str]
    description: Optional[str]
    logo_path: Optional[str]
    logo_url: Optional[str]

    email: Optional[str]
    phone: Optional[str]
    website_url: Optional[str]
    whatsapp_number: Optional[str]

    show_phone: bool
    show_email: bool
    show_website: bool

    payment_qr_path: Optional[str]
    payment_qr_url: Optional[str]
    payment_note: Optional[str]
    bank_account_name: Optional[str]
    bank_name: Optional[str]
    bank_account_number: Optional[str]
    bank_ifsc_swift: Optional[str]
    bank_reference_note: Optional[str]

    created_at: datetime
    updated_at: datetime
    default_currency: Optional[str] = None
    accepted_currencies: Optional[str] = None

    class Config:
        from_attributes = True


class CompanyProfilePublic(BaseModel):
    """Schema for public-facing company profile (used in shared itinerary)"""
    company_name: Optional[str]
    tagline: Optional[str]
    description: Optional[str]
    logo_url: Optional[str]

    email: Optional[str] = None  # Only included if show_email is True
    phone: Optional[str] = None  # Only included if show_phone is True
    website_url: Optional[str] = None  # Only included if show_website is True

    payment_qr_url: Optional[str]
    payment_note: Optional[str]

    class Config:
        from_attributes = True
